### PR TITLE
Jenkins plugin enhancements and fixes

### PR DESCRIPTION
**Pull request by umeshwaghode**
This PR fixes PLUG-736, PLUG-622, PLUG-619, PLUG-618, PLUG-613, PLUG-396
* Suppress benign errors by default (for ex. duplicate scan or timeout error). This can be disabled by defined JVM property 'suppressBenignErrors=false'
* Special character validation for custom field
* Introduced presetId 0 that makes SAST use presetid of previous scan of that project. If it is a new project, preset gets defaulted at SAST to 'Checkmarx Default'
* Pipeline script can be configured with scaTeamId instead of scaTeamPath. scaTeamId takes precedence.
* CxOrigin value now contains Jenkin's Plugin Version

Tests:
1.	It now shows FAILURE with message 
a.	when job is cancelled from the SAST server  (Delete scan in SAST while plugin is waiting for results)
b.	when incremental scan results with ‘Incremental changes threshold cross 7%’ error  (Add more java files to your repo)
c.	When scan times-out for a new project and when global flag ‘continue when timeout’ flag is not selected
d.	When scan times-out for existing project and when global flag ‘continue when timeout’ flag is not selected ( In order for scan to be full scan, add more java files to the test repo so that scan will be forced)
2.	It now shows SUCCESS with original error message as well as indication that error is being suppressed for below cases
a.	When duplicate scan error occurs
b.	When  ‘Source folder is empty…..’ error occurs (exclude all source files of the repo being used for testing)
c.	When ‘No files to zip’ error occurs (exclude entire files and folder or disable source code management. In case of CLI, pass empty folder as -locationpath)
d.	When scan times-out for a new project and when global flag ‘continue when timeout’ flag is selected
e.	When scan times-out for existing project and when global flag ‘continue when timeout’ flag is selected
f.	 
3.	In all cases of error getting suppressed, Last scan results will be fetched (if available)
4.	Suppression of error is always enabled from the Jenkins plugin for dup scan, source folder is empty, no files to zip cases.  Timeout case is controlled by the flag. (In CLI, none of the error suppression will occur. That’s expected.)

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by ENCHALWARPRANJAL — COMMENTED** [review]
All changes look fine to me.

File: src/main/java/com/checkmarx/jenkins/CxScanBuilder.java [hunk 1]
```diff
@@ -1280,7 +1303,14 @@ private String getCxOriginUrl(EnvVars env, CxLoggerAdapter log) {
         }
         return originUrl;
     }
-
+    private Boolean verifyCustomCharacters(String inputString) {
+    	 Pattern pattern = Pattern.compile("(^([a-zA-Z0-9#._]*):([a-zA-Z0-9#._]*)+(,([a-zA-Z0-9#._]*):([a-zA-Z0-9#._]*)+)*$)");
+         Matcher match = pattern.matcher(inputString);
```
Comment (joevin-slq-ibp) [inline src/main/java/com/checkmarx/jenkins/CxScanBuilder.java:1308]: What if inputString is null here @ENCHALWARPRANJAL ?
`java.lang.NullPointerException
 	at java.util.regex.Matcher.getTextLength(Matcher.java:1283)
 	at java.util.regex.Matcher.reset(Matcher.java:309)
 	at java.util.regex.Matcher.<init>(Matcher.java:229)
 	at java.util.regex.Pattern.matcher(Pattern.java:1093)
 	at com.checkmarx.jenkins.CxScanBuilder.verifyCustomCharacters(CxScanBuilder.java:1308)
 	at com.checkmarx.jenkins.CxScanBuilder.resolveConfiguration(CxScanBuilder.java:1338)
 	at com.checkmarx.jenkins.CxScanBuilder.perform(CxScanBuilder.java:902)
 	at jenkins.tasks.SimpleBuildStep.perform(SimpleBuildStep.java:123)
 	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:99)
 	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:69)
 	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 	at java.lang.Thread.run(Thread.java:748)
 Finished: FAILURE`

Comment (ENCHALWARPRANJAL) [reply src/main/java/com/checkmarx/jenkins/CxScanBuilder.java:1308]: Hi [joevin-slq-ibp](https://github.com/joevin-slq-ibp) can you mention the steps to reproduce this issue.

Comment (joevin-slq-ibp) [reply src/main/java/com/checkmarx/jenkins/CxScanBuilder.java:1308]: @ENCHALWARPRANJAL : i suggest to talk with Yair from the EMEA support team, I've created the case [#143029](https://checkmarx.force.com/CheckmarxCustomerServiceCommunity/s/case/5004K00000NIGAh/) with a Jenkinsfile to reproduce it.

**Comment by basil** [comment]
Hi @umeshwaghode and @ENCHALWARPRANJAL, sorry for commenting in an unrelated PR but I didn't know of any other way to reach you. I wanted to draw your attention to [JENKINS-69990](https://issues.jenkins.io/browse/JENKINS-69990). Let me know if I can help in any way!

**Comment by umeshwaghode** [comment]
> Hi @umeshwaghode and @ENCHALWARPRANJAL, sorry for commenting in an unrelated PR but I didn't know of any other way to reach you. I wanted to draw your attention to [JENKINS-69990](https://issues.jenkins.io/browse/JENKINS-69990). Let me know if I can help in any way!

Thanks for reaching out. I have access to JIRA. We can take the communication on this forward through the JIRA issues.

**Pull request merged**